core-services/dashboard-analytics/src/main/java/com/tarento/analytics/handler/MetricsInsightsHandler.java

```diff
@@ -64,8 +69,18 @@ public AggregateDto getInsights(AggregateDto aggregateDto, String visualizationC
 			if (textToDisplay == null) {
 				return aggregateDto;
 			} else {
-				textToDisplay = textToDisplay.replace(INSIGHT_INTERVAL_PLACEHOLDER,
-						insightsConfig.getInsightInterval());
+				if(difference == 0 ){
```

Comment:
@ajay-egov Please check if we can generalise the logic to populate the no difference text. A map with the list of labels based on intervals should work out